packages/invoices/src/OutboundShipment/DetailView/modals/hooks/usePackSizeController.tsx

```diff
@@ -38,23 +39,24 @@ const usePackSizeOptions = (
     );
   }, [packSizes]);
 
-  return options;
+  return { options, packSizes };
 };
 
 export const usePackSizeController = (lines: DraftOutboundLine[]) => {
-  const packSizes = useMemo(() => distinctSortedPackSizes(lines), [lines]);
-  const options = usePackSizeOptions(packSizes);
+  const { options, packSizes } = usePackSizes(lines);
 
-  const [selected, setSelected] = useState({ label: '', value: -1 });
+  const [selected, setSelected] = useState({ label: 'label.any', value: -1 });
 
-  const setPackSize = (newValue: number) => {
-    const packSizeOption = options.find(({ value }) => value === newValue);
-    if (!packSizeOption) return;
-    setSelected(packSizeOption);
-  };
+  const setPackSize = useCallback(
+    (newValue: number) => {
+      const packSizeOption = options.find(({ value }) => value === newValue);
+      if (!packSizeOption) return;
+      setSelected(packSizeOption);
+    },
+    [options, setSelected]
+  );
 
   useEffect(() => {
-    if (selected.value !== 0) return;
```

Comment:
This was messing with the initial value.. the behaviour I think is correct now 🤔

Reply:
with this removed, I cannot change to `Any` when editing an item which is currently using pack size `1`

The effect should only run on initial load - if the Select is manually changed, this block should not change the selected value back again.. but does!

Reply:
Originally, I had the default value as `{label: '', value: 0}` and the effect selects `{label: 'label.any', value: -1}`. Now that the default is changed, this line would need to change - or a different way to detect initial state